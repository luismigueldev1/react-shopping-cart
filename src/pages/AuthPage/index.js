import React, { useContext, useState } from "react";
import RegisterForm from "../../components/RegisterForm/";
import { StoreContext } from "../../context/StoreContext";
import { storeTypes } from "../../types/storeTypes";

export default function AuthPage({ history }) {
  const [isRegister, setIsRegister] = useState(false);
  const store = useContext(StoreContext);

  const handleIsRegister = () => {
    setIsRegister(!isRegister);
  };

  const handleIsInvited = () => {
    store.dispatch({
      type: storeTypes.login,
      payload: {
        email: "invitado",
      },
    });
    history.push("/checkout");
  };

  return (
    <div className="container">
      <div className="row justify-content-center align-items-center">
        <div className="col-sm-10 col-md-4  d-flex flex-column mt-5">
          <h3 className="text-center">Autenticaci√≥n</h3>
          {!isRegister ? (
            <>
              <button
                className="btn btn-primary  mt-3"
                onClick={handleIsRegister}
              >
                Registrarse
              </button>
              <button
                className="btn btn-secondary  mt-3"
                onClick={handleIsInvited}
              >
                Invitado
              </button>
            </>
          ) : (
            <RegisterForm setIsRegister={setIsRegister} />
          )}
        </div>
      </div>
    </div>
  );
}
