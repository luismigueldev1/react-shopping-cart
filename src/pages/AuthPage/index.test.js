import AuthPage from ".";
import { mount } from "enzyme";

describe("Pruebas en <AuthPage/>", () => {
  const wrapper = mount(<AuthPage />);
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
