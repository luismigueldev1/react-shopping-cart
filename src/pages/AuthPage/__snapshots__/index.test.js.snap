// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AuthPage/> debe mostrarse correctamente 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center align-items-center"
  >
    <div
      className="col-4  d-flex flex-column mt-5"
    >
      <h3
        className="text-center"
      >
        Autenticaci√≥n
      </h3>
      <button
        className="btn btn-primary  mt-3"
        onClick={[Function]}
      >
        Registrarse
      </button>
      <button
        className="btn btn-secondary  mt-3"
        onClick={[Function]}
      >
        Invitado
      </button>
    </div>
  </div>
</div>
`;
